oneapp-server/dao/ApplicationItemsDao.js

```diff
@@ -0,0 +1,46 @@
+const { SQLDataSource } = require('datasource-sql');
+const oracledb = require('oracledb');
+const { OneAppError } = require('../utils/OneAppError');
+
+class ApplicationItemsDao extends SQLDataSource {
+  async getItems(APPLICATION_ID, LANGUAGE) {
+    const con = await this.knex.client.pool.acquire().promise;
+    const response = await this.knex.client.transaction(async (tx) => {
+      const bindVars = {
+        app_id: { dir: oracledb.BIND_IN, val: APPLICATION_ID },
+        lang: { dir: oracledb.BIND_IN, val: LANGUAGE },
+        info: { dir: oracledb.BIND_OUT, type: 'OA_TT_APPLICATION_ITEMS' },
+      };
+      return tx.raw('begin OA_PKG_APP.SP_SELECT_APPLICATION_ITEMS(:app_id, :lang, :info); end;', bindVars);
+    }, { connection: con });
+    return response[0].getElement(0);
+  }
+
+  async updateContact(APPLICATION_NUMBER, input) {
+    const con = await this.knex.client.pool.acquire().promise;
+    const response = await this.knex.client.transaction(async (tx) => {
+      const ItemType = await con.getDbObjectClass('OA_TT_APPLICATION_ITEMS');
+      const item = new ItemType({
+        APPLICATION_NUMBER,
+        CODE_TYPE: 'FI', // Always used for food stamp
+        CODE: 'FI',
+        HAVE_THIS: input.HAVE_THIS ? 'Y' : 'N',
+      });
+      const bindVars = {
+        input: { dir: oracledb.BIND_IN, val: item },
+        msg: { dir: oracledb.BIND_OUT, type: oracledb.NUMBER },
+      };
+      return tx.raw('begin OA_PKG_APP.SP_UPDATE_APPLICATION_ITEMS(:input, :msg); end;', bindVars);
```

Comment:
This just denotes in the database that a SNAP application is being made, correct?

Reply:
correct. I generalized it below so that it can used in the future for all item types

Reply:
++ on generalizing. We'll just want to make sure we document the value needed to indicate it's a SNAP application in the static docs for @mikepif, otherwise I'm sure it won't be obvious.